np.sh

```diff
@@ -1,8 +1,18 @@
 #!/usr/bin/env sh
+
+if test 0 -ne `git status --porcelain | wc -l`; then
+	echo "Unclean working tree. Commit or stash changes first." >&2;
+	exit 128;
+fi
+
+if test "00" -ne `git fetch; git rev-list --count --left-right @'{u}'...HEAD | awk '{ print $1$2 }'`; then
```

Comment:
I think the `!=` check should be fine, and I do think option 2 is nicer, as it gives more information about what's wrong, to futher simplify the expression:

```sh
if ! git fetch --quiet 2>/dev/null; then
  echo problemz;
  exit 128;
fi
```